asdf/block.py

```diff
@@ -867,7 +870,7 @@ def _set_checksum(self, checksum):
 
     def _calculate_checksum(self, data):
         m = hashlib.new('md5')
-        m.update(self.data)
+        m.update(np.ascontiguousarray(self.data))
```

Comment:
It actually might make more sense to just to `m.update(self.data.flatten())` here.